src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -146,25 +183,51 @@ private static void InitializePortManagerClaimFiles(string traceAgentDirectory)
                 Directory.CreateDirectory(portManagerDirectory);
             }
 
-            var currentProcess = Process.GetCurrentProcess();
-            _processName = currentProcess.ProcessName.ToLowerInvariant();
-            _processId = currentProcess.Id;
-            var fileClaim = Path.Combine(portManagerDirectory, _processId.ToString());
+            var fileClaim =
+                Path.Combine(
+                    portManagerDirectory,
+                    $"{DomainMetadata.ProcessId}_{DomainMetadata.AppDomainId}");
 
             var portManagerFiles = Directory.GetFiles(portManagerDirectory);
-            var deleted = 0;
             if (portManagerFiles.Length > 0)
             {
-                var activePids = Process.GetProcesses().Select(p => p.Id.ToString()).ToList();
+                int? GetProcessIdFromFileName(string fullPath)
+                {
+                    var fileName = Path.GetFileNameWithoutExtension(fullPath);
+                    if (int.TryParse(fileName?.Split('_')[0], out var pid))
+                    {
+                        return pid;
+                    }
+
+                    return null;
+                }
+
                 foreach (var portManagerFileName in portManagerFiles)
                 {
                     try
                     {
-                        var claimPid = Path.GetFileName(portManagerFileName);
-                        if (!activePids.Contains(claimPid))
+                        var claimPid = GetProcessIdFromFileName(portManagerFileName);
+                        var isActive = false;
+
+                        try
+                        {
+                            if (claimPid != null)
+                            {
+                                using (var process = Process.GetProcessById(claimPid.Value))
+                                {
+                                    if (!process.HasExited)
```

Comment:
Checking `process.HasExited` is redundant. `Process.GetProcessById()` will only return a running process.